search-app/src/cmr/search/site/static.clj

```diff
@@ -12,23 +17,71 @@
   (static/generate
    "resources/public/site/docs/search/api.html"
    "templates/search-docs-static.html"
-   (merge
-    (data/base-static)
-    {:site-title "CMR Search"
-     :page-title "API Documentation"
-     :page-content (static/md-file->html "docs/api.md")})))
+   (merge (data/base-page {:cmr-application :search
+                           :execution-context :cli})
+          {:site-title "CMR Search"
+           :page-title "API Documentation"
+           :page-content (static/md-file->html "docs/api.md")})))
 
 (defn generate-site-docs
   "Generate CMR Search docs for routes and web resources."
   []
   (static/generate
    "resources/public/site/docs/search/site.html"
    "templates/search-docs-static.html"
-   (merge
-    (data/base-static)
-    {:site-title "CMR Search"
-     :page-title "Site Routes & Web Resource Documentation"
-     :page-content (static/md-file->html "docs/site.md")})))
+   (merge (data/base-page {:cmr-application :search
+                           :execution-context :cli})
+          {:site-title "CMR Search"
+           :page-title "Site Routes & Web Resource Documentation"
+           :page-content (static/md-file->html "docs/site.md")})))
+
+(defn generate-directory-resources
```

Comment:
doc strings

Reply:
NIT: I think this is pretty readable, but its getting a bit long.  It may be worth breaking up.

Reply:
Agreed. I was particularly unhappy with that function. I had it split up, but the sequencing was too awkward.

/me looks at it again ...

Okay, I think a different way of splitting it will work here. I'll give it a shot ...

Reply:
Nice. This turned out really well. The next push will have no docstrings; the one after that will :-)

Thanks!